src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributeList.cs

```diff
@@ -197,9 +199,17 @@ public bool TryGetAttributes(string name, out IEnumerable<TAttribute> attributes
                 throw new ArgumentNullException(nameof(name));
             }
 
-            attributes = Attributes.Where(attribute => NameEquals(name, attribute));
+            var matchedAttributes = new List<TAttribute>();
+            for (var i = 0; i < Attributes.Count; i++)
+            {
+                if (NameEquals(name, Attributes[i]))
+                {
+                    matchedAttributes.Add(Attributes[i]);
+                }
+            }
+            attributes = matchedAttributes;
 
-            return attributes.Any();
+            return matchedAttributes.Count > 0;
```

Comment:
The other consideration here if this was a super commonly used API is that you could return a struct here.